lib/output-file/index.js

```diff
@@ -1,36 +1,29 @@
 'use strict'
 
-const u = require('universalify').fromCallback
-const fs = require('graceful-fs')
+const u = require('universalify').fromPromise
+const fs = require('../fs')
 const path = require('path')
 const mkdir = require('../mkdirs')
 const pathExists = require('../path-exists').pathExists
 
-function outputFile (file, data, encoding, callback) {
-  if (typeof encoding === 'function') {
-    callback = encoding
-    encoding = 'utf8'
-  }
-
+async function outputFile (file, data, encoding = 'utf-8') {
   const dir = path.dirname(file)
-  pathExists(dir, (err, itDoes) => {
-    if (err) return callback(err)
-    if (itDoes) return fs.writeFile(file, data, encoding, callback)
 
-    mkdir.mkdirs(dir, err => {
-      if (err) return callback(err)
+  const itDoes = await pathExists(dir)
```

Comment:
Let's rename this to something sensible like `dirExists`, while we're here :grinning: